tests/provision/provisioning_test.go

```diff
@@ -0,0 +1,645 @@
+package tests
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/reporters"
+	. "github.com/onsi/gomega"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/torpedo/drivers/node"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/drivers/volume"
+	. "github.com/portworx/torpedo/tests"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	defaultVstate = 1
+)
+
+func TestVps(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	var specReporters []Reporter
+	junitReporter := reporters.NewJUnitReporter("/testresults/junit_basic.xml")
+	specReporters = append(specReporters, junitReporter)
+	RunSpecsWithDefaultAndCustomReporters(t, "Torpedo : Provisioning", specReporters)
+}
+
+var (
+	specNameRegex = regexp.MustCompile("{VPS_NAME}")
+	volKeyRegex   = regexp.MustCompile("{VOL_KEY}")
+	volLabelRegex = regexp.MustCompile("{VOL_LABEL}")
+	k8sCore       = core.Instance()
+)
+
+var _ = BeforeSuite(func() {
+	InitInstance()
+})
+
+// This test performs VolumePlacementStrategy's replica affinity  of application
+// volume
+var _ = Describe("{ReplicaAffinity}", func() {
+	var contexts []*scheduler.Context
+	It("has to schedule app and verify the volume replica affinity", func() {
+
+		var vpsSpec string
+		vpsRules := GetVpsRules(1)
+
+		for vkey, vrule := range vpsRules {
+			contexts = make([]*scheduler.Context, 0)
+			var lblData []labelDict
+			var setLabels int
+			Step("get nodes and set labels: "+vkey, func() {
+				lblData, setLabels = getTestLabels(vrule.GetLabels)
+				RemoveNodeLabels(lblData)
+				if setLabels == 1 {
+					lblnode := SetNodeLabels(lblData)
+					logrus.Debug("Nodes containing label", lblnode)
+					Expect(lblnode).NotTo(BeEmpty())
+				}
+			})
+
+			Step("rules of volume placement: "+vkey, func() {
+				vpsSpec = getVpsSpec(vrule.GetSpec)
+			})
+
+			Step("launch application with new vps specs :"+vkey, func() {
+				applyVpsSpec(vpsSpec)
+				logrus.Debugf("Spec Dir to rescan: %v", Inst().SpecDir)
+				Inst().S.RescanSpecs(Inst().SpecDir)
+
+				// Set VPS enabled flag
+				VpsMap := &scheduler.VpsParameters{
+					Enabled: true,
+					Vstate:  defaultVstate,
+				}
+				for i := 0; i < Inst().ScaleFactor; i++ {
+
+					ctxs, err := Inst().S.Schedule(fmt.Sprintf("replicaaffinity-%d", i), scheduler.ScheduleOptions{
+						AppKeys:            Inst().AppList,
+						StorageProvisioner: Inst().Provisioner,
+						ConfigMap:          Inst().ConfigMap,
+						VpsParameters:      VpsMap,
+					})
+					Expect(err).NotTo(HaveOccurred())
+					Expect(ctxs).NotTo(BeEmpty())
+					contexts = append(contexts, ctxs...)
+				}
+
+				for _, ctx := range contexts {
+					ValidateContext(ctx)
+				}
+
+			})
+
+			opts := make(map[string]bool)
+			opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+
+			vrule.CleanVps() //TODO: function arg for cleaning up the testcase environment
```

Comment:
CleanVps not needed. The cleanup should be generic where it removes all labels with a known prefix